portal-web/test/functional/com/liferay/portalweb/tests/enduser/frontendinfrastructure/uiinfrastructure/SPA.testcase

```diff
@@ -33,4 +33,86 @@ definition {
 		AssertVisible.assertVisibleNoSPAWait(locator1 = "//html[contains(@class,'lfr-spa-loading')]");
 	}
 
+	@description = "Validate user can configure SPA timeout notification"
+	@priority = "4"
+	test ConfigureSPANotificationTimeout {
+		property test.name.skip.portal.instance = "SPA#AssertLoadingBar";
+
+		var appName = "Liferay Frontend JS SPA Web";
+
+		task ("Configure User Notification Timeout to 1 second") {
+			ApplicationsMenu.gotoPortlet(
+				category = "Configuration",
+				panel = "Control Panel",
+				portlet = "System Settings");
+
+			SystemSettings.gotoConfiguration(
+				configurationCategory = "Infrastructure",
+				configurationName = "Frontend SPA Infrastructure",
+				configurationScope = "System Scope");
+
+			Type(
+				key_text = "User Notification Timeout",
+				locator1 = "TextInput#ANY",
+				value1 = "1000");
+
+			SystemSettings.saveConfiguration();
+		}
+
+		task ("Reset the SPA module in App Manager to fully apply the SPA configuration") {
+			Navigator.openURL();
+
+			ApplicationsMenu.gotoPortlet(
+				category = "System",
+				panel = "Control Panel",
+				portlet = "App Manager");
+
+			AppManager.viewAppCP(
+				appName = "${appName}",
+				appStatus = "Active",
+				searchTerm = "SPA");
+
+			AppManager.deactivateAppCP(appName = "${appName}");
+
+			AppManager.viewAppCP(
+				appName = "${appName}",
+				appStatus = "Resolved",
+				searchTerm = "SPA");
+
+			AppManager.activateAppCP(appName = "${appName}");
+
+			AppManager.viewAppCP(
+				appName = "${appName}",
+				appStatus = "Active",
+				searchTerm = "SPA");
+		}
+
+		task ("Validate time out warning by adding a virtual instance") {
```

Comment:
This is not deterministic. What if all of a sudden `adding a virtual instance` is so fast that it takes less than our timeout? This is kinda a way to prevent teams to improving that further 😂 

IMHO, as we've discussed other times, you should find a way to make sure that your test always reproduces and has no dependencies out of your control. 

The more direct way I can think of to do this would be to somehow create a `SlowPortlet` that would take longer than a second to render (`Thread.sleep`, maybe) and then have it in the page you want to navigate to.

Other approaches would be to forcibly delay the requests by wrapping the native `fetch` in a `setTimeout`, but this is a leaky and depends on our internal implementation so it's not as safe as the other approach.

I'm sure there are other ways and this is something I think QA should have already available to mimic these kind of situations rather than executing real long-standing actions.

Reply:
> Another way is to have the page execute some long-running groovy (or other) script that you control how long it takes.
I can update the test to go to Server Administration > Script and execute sleep(5000).